### DISCUSSION TITLE
Flow for testing built docker images?

### DISCUSSION THREAD

**Discussion by marverix** [docker/build-push-action#1104] [General]
Hello,

I'm writing GH Workflows for the company where I'm employed. In Jenkins we had such flow that we first built a docker image, next we tested it with [dgoss](https://github.com/goss-org/goss) (dgoss is docker wrapper for goss), and after that we pushed do DockerHub. So I'm looking for available actions on markerplace, and I'm wondering if this action support such flow? I thought that maybe I can use this action twice - once for building, and once for pushing, but from what I can see is that this action only support disabling pushing (so I can do "build only" part) but does not support disabling building (so I can do "push only" part).

**Comment by crazy-max** [comment]
Yes we have this kind of flow in ours docs: https://docs.docker.com/build/ci/github-actions/test-before-push/

**Reply by marverix** [reply]
Super! Thanks! But does it build twice or once? I don't see any flag to disable second build. Is it using local cache to omit second build?

**Reply by crazy-max** [reply]
Yes it's using local cache of previous one

**Reply by marverix** [reply]
Thanks
